modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache

```diff
@@ -4,6 +4,77 @@
 
 export const BASE_PATH = "{{{basePath}}}".replace(/\/+$/, "");
 
+export interface ConfigurationParameters {
+    basePath?: string; // override base path
+    fetchApi?: FetchAPI; // override for fetch implementation
+    middleware?: Middleware[]; // middleware to apply before/after fetch requests
+    queryParamsStringify?: (params: HTTPQuery) => string; // stringify function for query strings
+    username?: string; // parameter for basic security
+    password?: string; // parameter for basic security
+    apiKey?: string | ((name: string) => string); // parameter for apiKey security
+    accessToken?: string | Promise<string> | ((name?: string, scopes?: string[]) => string | Promise<string>); // parameter for oauth2 security
+    headers?: HTTPHeaders; //header params we want to use on every request
+    credentials?: RequestCredentials; //value for the credentials param we want to use on each request
+}
+
+export class Configuration {
+    constructor(private configuration: ConfigurationParameters = {}) {}
+
+    set defaultConfig(configuration: Configuration) {
```

Comment:
I would suggest to rename this to `set config` or similar to avoid weird cases where a custom config is used and you need to call `defaultConfig`to change the configuration.

Reply:
@TiFu 
Thank you for your review!
I also felt need to avoid such cases, so we have made the correction at e8e14d6 .